Examples/Python/include/Acts/Plugins/Python/Utilities.hpp

```diff
@@ -64,5 +69,26 @@ void patchKwargsConstructor(T& c) {
   }                              \
   do {                           \
   } while (0)
+  
 
-}  // namespace Acts::Python
\ No newline at end of file
+/// This macro is needed to use the BOOST_PP_SEQ_FOR_EACH loop macro
+#define ACTS_PYTHON_MEMBER_LOOP(r, data, elem) ACTS_PYTHON_MEMBER(elem);
+
+/// A macro that uses Boost.Preprocessor to create the python binding for and algorithm and the additional config struct.
+#define ACTS_PYTHON_DECLARE_ALGORITHM(algorithm, name, ...)                 \
+  {                                                                         \
+    using Alg = algorithm;                                                  \
+    using Config = Alg::Config;                                             \
+    auto alg =                                                              \
+        py::class_<Alg, ActsExamples::BareAlgorithm, std::shared_ptr<Alg>>( \
+            mex, name)                                                      \
+            .def(py::init<const Config&, Acts::Logging::Level>(),           \
+                 py::arg("config"), py::arg("level"))                       \
+            .def_property_readonly("config", &Alg::config);                 \
+                                                                            \
+    auto c = py::class_<Config>(alg, "Config").def(py::init<>());           \
+    ACTS_PYTHON_STRUCT_BEGIN(c, Config);                                    \
+    BOOST_PP_SEQ_FOR_EACH(ACTS_PYTHON_MEMBER_LOOP, _,                       \
+                          BOOST_PP_VARIADIC_TO_SEQ(__VA_ARGS__))            \
```

Comment:
I was originally trying to do something like this, but didn't want to use boost for the functionality. But I guess this is a good idea since it simplifies the code.

Reply:
Yeah but I also quickly thought about doing this without boost, but after seeing what obscure constructs are needed for a preprocessor loop...^^